includes/Hooks.php

```diff
@@ -87,7 +87,49 @@ public static function onBaseTemplateToolbox(\BaseTemplate &$baseTemplate, array
 			$toolbox['usercomments']['id'] = 't-usercomments';
 		}
 	}
+	
+	public static function onSidebarBeforeOutput(\Skin $skin, &$sidebar) {
+		if ( version_compare( MW_VERSION, '1.35', '<' ) ) {
+			return false;
+		}
+		
+		$commentAdmin = $skin->getUser()->isAllowed('commentadmin-restricted');
+		$user = $skin->getRelevantUser();
+		
+		if ($user && $commentAdmin) {
+				$sidebar['TOOLBOX'][] = [
+				'text' => wfMessage('sidebar-usercomments')->text(),
+				'href' => \SpecialPage::getTitleFor('FlowThreadManage')->getLocalURL(array(
+					'user' => $user->getName(),
+				)),
+			];
+		}
+	}
+	
+	public static function onSkinTemplateNavigation_Universal(\SkinTemplate $skinTemplate, array &$links) {
+		if ( version_compare( MW_VERSION, '1.35', '<' ) ) {
+			return false;
+		}
+		
+		$commentAdmin = $skinTemplate->getUser()->isAllowed('commentadmin-restricted');
+		$user = $skinTemplate->getRelevantUser();
 
+		$title = $skinTemplate->getRelevantTitle();
+		if (Helper::canEverPostOnTitle($title) && ($commentAdmin || Post::userOwnsPage($skinTemplate->getUser(), $title))) {
+			// add a new action
+			$links['actions']['flowthreadcontrol'] = [
+				'id' => 'ca-flowthreadcontrol',
+				'text' => wfMessage('action-flowthreadcontrol')->text(),
+				'href' => \SpecialPage::getTitleFor('FlowThreadControl', $title->getPrefixedDBKey())->getLocalURL()
+			];
+		}
+
+		return true;
+	}
+	
+	//Deprecated in MW 1.35+. See https://www.mediawiki.org/wiki/Manual:Hooks/SkinTemplateOutputPageBeforeExec
+	//Use onSkinTemplateNavigation_Universal instead.
+	
```

Comment:
Is there a reason not to remove onSkinTemplateOutputPageBeforeExec all together?